bclconvert.wdl

```diff
@@ -19,189 +15,169 @@ struct FastqCollection {
+    String modules = "bclconvert-scripts/1.0"
+    String samplesheetScript = "$BCLCONVERT_SCRIPTS_ROOT/bin/buildSamplesheet.py"
+    Int memory = 4
+    Int timeout = 2
+  }
+
+  parameter_meta {
+    sample: "Object which holds a metadata for a sample"
+    samplesheetScript: "Script for generating sample sheet"
+    lanes: "Lanes to extract"
+    basesMask: "An Illumina bases mask string to use. If absent, the one written by the instrument will be used."
+    memory: "Memory allocated for running this task. Default 4"
+    modules: "Modules for running bclconvert task"
+    timeout: "Timeout for building a samplesheet. Default 2"
+  }
 
   command <<<
-    python3 <<CODE
-    import json
-    with open("samplesheet.csv", "w") as ss:
-      ss.write("[Data]\n")
-      ss.write("Sample_ID,index,index2\n")
-      with open("~{write_json(samples)}") as js:
-        d=json.load(js)
-        print(d)
-        for sample in d['samples']:
-           name=sample['name']
-           barcodes=sample['barcodes']
-           for barcode in barcodes:
-             (bc1,bc2)=barcode.split("-")
-             ss.write("%s,%s,%s\n" %(name,bc1,bc2))
-    ss.close()
-    CODE
+    python3 ~{samplesheetScript} -i "~{write_json(sample)}" -m "~{basesMask}" -l "~{sep=',' lanes}"
   >>>
 
+  runtime {
+      memory: "~{memory}G"
+      modules: "~{modules}"
+      timeout: "~{timeout}"
+  }
+
   output {
     File samplesheet = "samplesheet.csv"
   }
 }
 
-
-task bclconvert {
+# =====================
+# Run bclconvert on HPC
+# =====================
+task runBclconvertHpc {
   input {
     String runFolder
-    String runName
+    String runName = basename(runFolder)
     File sampleSheet
     Boolean firstTileOnly = false
     Boolean noLaneSplitting = false
-    Boolean onlyMatchedReads = false
+    Boolean onlyMatchedReads = true
     String fastqCompression = 'gzip'
     Int fastqCompressionLevel = 1
     Int timeout = 40
     Int memory = 32
-	String modules
+    String modules = "bclconvert-scripts/1.0 bclconvert/4.2.7-2"
+    String bclconvertScript = "$BCLCONVERT_SCRIPTS_ROOT/bin/runBclconvert.py"
+    String? additionalParameters
+  }
+
+  parameter_meta {
+    runFolder: "Run folder"
+    runName: "Run name"
+    sampleSheet: "File with sample sheet"
+    firstTileOnly: "Flag for processing first tile only. Default false"
+    noLaneSplitting: "Flag to disable lane splitting. Default false"
+    onlyMatchedReads: "Process only matched reads. Default true"
+    fastqCompression: "Compression type of fastq files. Default gzip"
+    fastqCompressionLevel: "Fastq compression level. Default 1"
+    bclconvertScript: "Script for generating sample sheet"
+    timeout: "Timeout for this task. Default 40"
+    memory: "Memory allocated for running this task. Default 32"
+    modules: "Modules for running bclconvert task"
+    additionalParameters: "Pass parameters which were not exposed"
   }
 
   command <<<
+  set -euo pipefail
   bcl-convert -f \
```

Comment:
I think you want to run this with `--bcl-only-matched-reads`:
```
  --bcl-only-matched-reads arg           For pure BCL conversion, do not output files for 'Undetermined' [unmatched] 
                                         reads (output by default)
```

Reply:
Both flavours are using _--bcl-only-matched-reads_ set to true by default, not sure what you mean